packages/devextreme/js/__internal/core/state_manager/utils.ts

```diff
@@ -0,0 +1,108 @@
+export function splitStatePath(statePath: string, separator = '.'): string[] {
+  if (!statePath || typeof statePath !== 'string') {
+    return [];
+  }
+
+  return statePath.split(separator).filter(Boolean);
+}
+
+export function isValidStatePath(statePath: string): boolean {
+  const parts = splitStatePath(statePath);
+  return parts.length >= 2;
+}
+
+export function areEqual(inputA: unknown, inputB: unknown): boolean {
+  function iter(a: unknown, b: unknown, visited = new WeakMap()): boolean {
+    if (a === b) return true;
+
+    if (a === null || b === null || a === undefined || b === undefined) {
+      return a === b;
+    }
+
+    if (typeof a !== 'object' || typeof b !== 'object') {
+      return a === b;
+    }
+
+    const visitedA = visited.get(a);
```

Comment:
Naming is confusing, I guess, it's `b`:
```suggestion
    const visitedB = visited.get(a);
```

Reply:
I removed the `areEqual` function for now. 
I noticed that, for now, we can rely on the responsibility of a value container (preact/signal will do it) to trigger changes only when the current value differs from the previous value.
I suggest we postpone searching for a library until we need it.